### PR TITLE
Task/transaction id header

### PR DESCRIPTION
**Pull request by dmoranj**
Implements #288

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -98,6 +98,26 @@ iotagentLib.statsRegistry.add('statName', statIncrementalValue, callback)
...
+requests entering both from the Northbound and the Southbound.
+
+To follow the transactions, a new Domain is created for each incoming request; in the case of Northbound requests, this
+domain is automatically created by a Express middleware, and no further action is needed from the user. For the case of
+Southbound requests, the user is responsible of creating an stopping the transaction, using the `ensureSouthboundDomain`
+and `finishSouthBoundTransaction`. In this case, the transaction will last from the invocation to the formar to the
+invocation of the latter.
+
+The Transaction Correlator is used along all the IoT Platform to follow the trace of a transaction between multiple components.
+To do so, in all the HTTP requests sent to other components of the platform, a custom header named `` is sent with the
```
Comment (anabelengp) [inline README.md]: the doc can already be updated with `Fiware-Transaction` name for the header

File: README.md [hunk 2]
```diff
@@ -98,6 +98,26 @@ iotagentLib.statsRegistry.add('statName', statIncrementalValue, callback)
...
+correlator of the transaction that generated the request. If a component of the platform receives a request containing
+this header that start a transaction, the component will create the transaction with the received correlator, instead
```
Comment (anabelengp) [inline README.md]: trivial (syntax): starts a transaction

**Comment by anabelengp** [comment]
LGTM (minor comments in docs)

**Comment by AlvaroVega** [comment]
Are you logging this new header in your component traces?

**Comment by dmoranj** [comment]
@AlvaroVega Yes, I am logging them automatically, adding it to the `reqDomain.trans` field (domain attributes are automatically logged using logops).

**Comment by dmoranj** [comment]
Typos fixed in 5ee685e

File: README.md [hunk 3]
```diff
@@ -98,6 +98,26 @@ iotagentLib.statsRegistry.add('statName', statIncrementalValue, callback)
...
+requests entering both from the Northbound and the Southbound.
+
+To follow the transactions, a new Domain is created for each incoming request; in the case of Northbound requests, this
+domain is automatically created by a Express middleware, and no further action is needed from the user. For the case of
+Southbound requests, the user is responsible of creating an stopping the transaction, using the `ensureSouthboundDomain`
+and `finishSouthBoundTransaction`. In this case, the transaction will last from the invocation to the formar to the
+invocation of the latter.
+
+The Transaction Correlator is used along all the IoT Platform to follow the trace of a transaction between multiple components.
+To do so, in all the HTTP requests sent to other components of the platform, a custom header named `Fiware-Transaction` is sent with the
```
Comment (gtorodelvalle) [inline README.md]: Fiware-Transaction -> Fiware-Correlator

**Comment by gtorodelvalle** [comment]
LGTM after the previous minor comment... :)

**Pull request merged**